### PR TITLE
Fix #3095: Add tls_verify and tls_cacerts to support self-signed servers

### LINKED ISSUE #3095: Regression 1.4.8 -> 1.5a2: intersphinx not honoring ssl certs anymore
I have a local PyPI repo (devpi server) hosting wheels and docs for two projects `my-project` and `another-project`, the apidocs of `another-project` reference API of `my-project`. A local SSL certificate for the repo domain is installed correctly.

Building with `Sphinx==1.4.8`:

```
$ python setup.py doc             
running build_doc
Running Sphinx v1.4.8
loading pickled environment... not yet created
loading intersphinx inventory from https://pypi.repo.local/prod/my-project/0.9.0/+doc/objects.inv...
building [mo]: all of 0 po files
building [latex]: all source files
updating environment: 6 added, 0 changed, 0 removed
reading sources... [100%] index                                                                                                                                                                                                                              
...
building [mo]: all of 0 po files
building [html]: all source files
updating environment: 0 added, 0 changed, 0 removed
looking for now-outdated files... none found
preparing documents... done
writing output... [100%] index                                                                                                                                                                                                                               
generating indices... genindex py-modindex
writing additional pages... search
copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 24 warnings.
```

When testing out `Sphinx==1.5a2`:

```
(buildenv) $ python setup.py doc
running build_doc
Running Sphinx v1.5a2
loading pickled environment... not yet created
loading intersphinx inventory from https://pypi.repo.local/prod/my-project/0.9.0/+doc/objects.inv...
WARNING: intersphinx inventory 'https://pypi.repo.local/prod/my-project/0.9.0/+doc/objects.inv' not fetchable due to <class 'requests.exceptions.SSLError'>: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:645)
loading intersphinx inventory from objects.inv...
WARNING: intersphinx inventory 'objects.inv' not fetchable due to <class 'FileNotFoundError'>: [Errno 2] No such file or directory: '/var/lib/jenkins/jobs/another-project-build/workspace/doc/source/objects.inv'
building [mo]: all of 0 po files
building [latex]: all source files
updating environment: 6 added, 0 changed, 0 removed
reading sources... [100%] index
...
```

Something has changed since v1.4.8 so the local ssl cert is not accepted by intersphinx anymore.

### PR DESCRIPTION
**Pull request by tk0miya**
This is a fix for #3095.
I would like to review before merging.
Thanks,

### CODE REVIEW

File: sphinx/util/requests.py [hunk 1]
```diff
@@ -53,3 +56,46 @@ def is_ssl_error(exc):
             return True
         else:
             return False
+
+
+def _get_tls_cacert(url, config):
```
Comment (shimizukawa) [inline sphinx/util/requests.py:62]: Please write docstring. The behavior of this function is bit complex.

File: sphinx/util/requests.py [hunk 2]
```diff
@@ -53,3 +56,46 @@ def is_ssl_error(exc):
...
+        return certs
+    else:
+        hostname = urlsplit(url)[1]
+        if '@' in hostname:
+            hostname = hostname.split('@')[1]
+
+        return certs.get(hostname, True)
+
+
+def get(url, **kwargs):
```
Comment (shimizukawa) [inline sphinx/util/requests.py:84]: docstring please.

File: sphinx/util/requests.py [hunk 3]
```diff
@@ -53,3 +56,46 @@ def is_ssl_error(exc):
...
+
+    with warnings.catch_warnings():
+        if not kwargs.get('verify'):
+            # ignore InsecureRequestWarning if verify=False
+            warnings.filterwarnings("ignore", category=InsecureRequestWarning)
+        return requests.get(url, **kwargs)
+
+
+def head(url, **kwargs):
+    kwargs.setdefault('headers', dict(useragent_header))
```
Comment (shimizukawa) [inline sphinx/util/requests.py:104]: docstring please.

**Comment by shimizukawa** [comment]
LGTM with nits :)

**Comment by tk0miya** [comment]
Thank you for reviewing. updated.
In addition, could you check this at your environment (if possible)?

**Comment by shimizukawa** [comment]
> In addition, could you check this at your environment (if possible)?

Done. It works!

**Comment by tk0miya** [comment]
Thank you for confirmation!

**Pull request merged**